Hardware/Hardware/Printer/AirPrintReceipt/AirPrintReceiptPrinterService.swift

```diff
@@ -38,3 +44,27 @@ public final class AirPrintReceiptPrinterService: PrinterService {
         }
     }
 }
+
+extension AirPrintReceiptPrinterService: UIPrintInteractionControllerDelegate {
+    public func printInteractionController(_ printInteractionController: UIPrintInteractionController, choosePaper paperList: [UIPrintPaper]) -> UIPrintPaper {
+        let pageSize = self.receiptContent?.preferredPageSizeForPrinting ?? AirPrintReceiptPrinterService.defaultReceiptPageSize
```

Comment:
I am curious, what is the content of `paperList` for the Brother printer, the one we support? According to the docs it should be: 
`An array of UIPrintPaper objects that represent combinations of paper sizes and imageable areas supported by the selected printer.`

I am curious to know how many different paper sizes it reports, for a printer that is meant to use roll paper (of fixed width)

Reply:
```
(lldb) po paperList.first?.paperSize
▿ Optional<CGSize>
  ▿ some : (289.13385826771656, 792.0)
    - width : 289.13385826771656
    - height : 792.0
 ```

This comes from a previous call to the other delegate method `printInteractionController(_ printInteractionController: UIPrintInteractionController, cutLengthFor paper: UIPrintPaper)`, using the constants in `ReceiptContent`.

Reply:
https://developer.apple.com/documentation/uikit/uiprintinteractioncontroller?changes=l_2&language=objc we are using `printPageRenderer` as the approach here. There is a `_printSettings.paper` private API property which shows `name: "na_letter_8.5x11in"` that has me a bit suspicious.

Reply:
Thanks @joshheald  Does paperList contain more than one item? When printInteractionController :choosePaper: get called does it get passed anything in paperList?

Reply:
No, just the one item in `paperList`. That example _was_ taken from `printInteractionController :choosePaper:` being called, is that where you meant to check, or is there somewhere else as well?

Reply:
Actually, the first time round it doesn't get a paperSize. It then gets called again with the size pasted above (which I think it got from the first call to this method. It's a bit odd.)

Reply:
That's on opening the print modal. Tapping print it got called again with that same 612x1008 size

Reply:
I'm not really sure why it's inconsistent in the sizes it reports 😬

Reply:
I'm not sure either. I'll document it and report it to Apple